### PR TITLE
Angular: Use Nx function to read non-angularCli configs

### PR DESCRIPTION
**Pull request by mandarini**
Issue: #12232

## What I did
This is a fix for a behavior described [here](https://github.com/nrwl/nx/issues/4305) and [here](https://github.com/storybookjs/storybook/issues/12232).

A new function is created in the `@nrwl/workspace` package, which will take care of reading/parsing the workspace configuration, and returning it to Storybook in the `angularCli` format, which it expects.

The new functionality will work after this [Nx PR](https://github.com/nrwl/nx/pull/4440) is successfully merged.

### CODE REVIEW

**Review by shilman — APPROVED** [review]
LGTM @mandarini and thanks so much for the contribution. i'm OK with it except for the comments above. will get a few more eyes on it and then merge/release if nobody objects. after you've tested the alpha, we can patch it back into 6.1.x!

**Comment by merceyz** [comment]
> @gaetanmaisse @merceyz should @nrwl/workspace be an optional peer dependency for yarn2 compat?

Based on its usage, yes, though not just for Yarn PnP (Yarn 2 != pnp as Yarn 2 also supports node_modules) it must be declared so package managers can account for it when they do hoisting

**Comment by mandarini** [comment]
I just updated my PR, added `@nrwl/workspace` as an optional peer dependency!

File: app/angular/package.json
Comment (merceyz) [inline app/angular/package.json:93]: Needs to be a devDependency as well so that it's accesible for the code

**Review by gaetanmaisse — APPROVED** [review]

**Review by gaetanmaisse — CHANGES_REQUESTED** [review]

**Comment by gaetanmaisse** [comment]
The failing unit test looks related to a caching/hoisting issue in the `node_modules`, I did an ssh connection to the CI container, `yarn install`, reran the tests and everything was ok. Maybe updating the `yarn.lock` will fix that.

File: app/angular/src/server/angular-cli_config.ts
Comment (ThibaudAV) [inline app/angular/src/server/angular-cli_config.ts:54]: I don't know if it's better (or if it's possible)🙈 . but an alternative without try/catch would be to let storybook choose if it should use nx or not 🤷‍♂️ 

Something like this: 
`const isNxWorkspace = fs.existsSync(path.join(dirToSearch, 'nx.json'))`

File: app/angular/package.json
Comment (mandarini) [reply app/angular/package.json:76]: Ok, I'll use `11.1.0-beta.2`! How can I capture the yarn.lock changes, though?

File: app/angular/src/server/angular-cli_config.ts
Comment (mandarini) [reply app/angular/src/server/angular-cli_config.ts:54]: The thing is that it will need to access the `readWorkspaceConfig()` from `@nrwl/workspace`, so I apologized for using the `eslint-disable` to let me require it... :)

File: app/angular/package.json
Comment (mandarini) [reply app/angular/package.json:76]: It still defaults to 
```"
@nrwl/workspace@>=11.1.0-beta.2":
  version "9999.0.0"
```
since `11.1.0-beta.2` is not an actual tag/release. We can wait until the end of the week, when that release will happen, maybe?

Comment (gaetanmaisse) [reply app/angular/package.json:76]: Yes sure, I let you update that when the release will be done and then ping me back 😉

Comment (mandarini) [reply app/angular/package.json:76]: @gaetanmaisse I committed the `yarn.lock` changes too!

**Review by Hasbiallah210012 — APPROVED** [review]
[]()

File: app/angular/src/server/angular-cli_config.ts
Comment (Kamarulzamansalehudin) [reply app/angular/src/server/angular-cli_config.ts:54]: Done

Comment (ThibaudAV) [reply app/angular/src/server/angular-cli_config.ts:54]: okay, 
my point is more about the try/catch. I find that the way we manage the nominal case (without nx) in a Catch is not pretty.
and I propose this solution if it works and suits you of course :) 

```js
const isNxWorkspace = fs.existsSync(path.join(dirToSearch, 'nx.json'))
if(isNxWorkspace) {
 return  require('@nrwl/workspace').readWorkspaceConfig({
      format: 'angularCli',
    });
}
... 

```

**Comment by mandarini** [comment]
Hi @gaetanmaisse ! Let me know if I need to make any further changes! I see that the tests are still failing :(

**Comment by gaetanmaisse** [comment]
We are currently working on fixing the tests (they are failing on `next` too), I will update your branch as soon as the fix will be merged 😉

**Comment by mandarini** [comment]
Thank you very much! :)

File: app/angular/src/server/angular-cli_config.ts
Comment (gaetanmaisse) [reply app/angular/src/server/angular-cli_config.ts:54]: @ThibaudAV are you ok to go with the current state?

**Review by gaetanmaisse — APPROVED** [review]

File: app/angular/src/server/angular-cli_config.ts
Comment (ThibaudAV) [reply app/angular/src/server/angular-cli_config.ts:54]: Yes 🚀 , I have plans to improve this part. I will do it with a dedicated PR  ;)

**Pull request merged**
